### PR TITLE
wip fix paste does not work in firefox

### PR DESCRIPTION
**Pull request by juanbercoff**
- #301 

This change use an event based approach instead of using the Clipboard API. This allows copying/pasting to be browser independent so now works on firefox/edge. 

It's WIP, if you like the approach I could polish the code and  handle some edge cases I noticed.

### CODE REVIEW

File: src/gridGL/QuadraticGrid.tsx [hunk 1]
```diff
@@ -9,18 +9,28 @@ import { CellInput } from './interaction/CellInput';
...
 import { PanMode } from '../atoms/gridInteractionStateAtom';
+import { ClipboardData, generateClipboardStrings, pasteFromTextOrHtml } from '../grid/actions/clipboard/clipboard';
+import { Coordinate } from './types/size';
 
 interface IProps {
   sheetController: SheetController;
   app: PixiApp;
 }
 
+export type CopiedData = {
```
Comment (davidkircos) [inline src/gridGL/QuadraticGrid.tsx:20]: move type declaration to clipboard file

File: src/gridGL/QuadraticGrid.tsx [hunk 2]
```diff
@@ -105,6 +115,42 @@ export default function QuadraticGrid(props: IProps) {
     app: props.app,
   });
 
+  useEffect(() => {
```
Comment (davidkircos) [inline src/gridGL/QuadraticGrid.tsx:118]: This approach seems to be more robust

**Comment by davidkircos** [comment]
@juanbercoff I like this approach. However, because the clipboard is kept in the sate, it will not work across sessions and browser tabs, will it?

**Comment by juanbercoff** [comment]
> @juanbercoff I like this approach. However, because the clipboard is kept in the state, it will not work across sessions and browser tabs for formatted cells, will it?

Yes, you are right. You can solve that using localforage, but I don't think it's worth it. A huge downside is that you can't copy structured content cross-browser/application. With the current implementation that's working.

The main issue with Firefox is that the Clipboard API is not fully supported. I'll try to find a workaround.

**Comment by davidkircos** [comment]
Maybe the best approach is an optimistic one? If we are on a supported
platform - then use the Clipboard API. If we are not - use your current
approach which should work on all platforms.

**Comment by davidkircos** [comment]
In Figma you can copy and paste structured content from the website version
to the electron version of the app. If the user is on Chrome we can do this
with the current approach.

**Comment by juanbercoff** [comment]
Ok, that should work.

**Pull request closed without merging**